### DISCUSSION TITLE
Effects on Hydrogen

### DISCUSSION THREAD

**Discussion by Morbott** [hydrogen-music/hydrogen#1147] [Hydrogen on Windows]
Hello... new here... I'd like to thank anyone in advance for helping me learn this software.
I'm a bit of a newb with this sort of thing, so I appreciate your patience.
What I'm wondering is if there are any effects that exist for Hydrogen, that I can use to help modify the sound of my drums?
Thinks like reverb or some distortion.
Maybe I'm thinking about this in the wrong way. If anyone can help point me in the right direction, I'd appreciate it a lot.
Oh, if it makes a difference, I'm using it on Windows.
Thank you,
M

**Comment by trebmuh** [comment]
Yes, you can add effects to the drum sounds using [the Hydrogen mixer](http://hydrogen-music.org/documentation/manual/manual_en.html#sect.mixer).

**Reply by Morbott** [reply]
Hi trebmuh,
Hmmm... seems like all the links listed in the docs are for Linux only.
I'm on Windows.
Darn.

**Reply by trebmuh** [reply]
They are Free and Open Source Software. Hence, they can be built on any platform.

There are some LADSPA plugins windows built on this [Audacity page](https://www.fosshub.com/Audacity-old.html).
I don't know how it works or how to install those, I'm only giving you this pointer in the hope it works for you.

Let us know how it goes.

**Reply by Morbott** [reply]
Thanks for the tips.
I'll take a look. Fingers crossed.
M

**Comment by Morbott** [comment]
I admit, I'm a little over my head with this.
If there are any Windows users who have any tips on getting FX into Hydrogen on Windows, I'd be very thankful.
M

**Reply by theGreatWhiteShark** [reply]
It's probably not the solution you are looking for but there are Linux distributions dedicated to audio/multimedia production, like [Ubuntu studio](https://ubuntustudio.org/) and [KXStudio](https://kx.studio/News/). No in depth knowledge about Linux required. Hydrogen, the audio server and many effects are already setup. Using dual boot you can run it next to your Windows distribution

**Comment by dfkettle** [comment]
Arriving a little late to the conversation, but are LADSPA effects even supported in the Windows version of Hydrogen? When I click on the Edit button for one of the four effects, I just get a pop-up window telling me "LADSPA effects are not available in this version of Hydrogen". And if they are, how do I point Hydrogen to the directory where they're installed? In other programs I can specify the directories where plug-ins have been installed. But I don't see this option anywhere in Hydrogen's Preferences menu.

![Hydrogen](https://user-images.githubusercontent.com/8409615/126872185-03a957db-e1c0-4704-ab63-f2234f4d6f9e.png)

**Reply by theGreatWhiteShark** [reply]
Hey @dfkettle ,

Some things did change in the Windows build lately and just from the looks of it it should be enabled now. (I'm not a Windows user and can't tell). Have you tried the [latest release](https://github.com/hydrogen-music/hydrogen/releases/tag/1.1.0-beta1)?

**Reply by dfkettle** [reply]
I installed v1.0.2 (which is supposed to be the "latest stable version"). I tried both the 32-bit and the 64-bit versions, but neither worked (as far as plug-ins are concerned, that is). Should I give the beta version a try?

**Reply by dfkettle** [reply]
I just installed the beta version that you pointed me to, but I still get the same error. Should I create an issue for this on Github?

**Reply by dfkettle** [reply]
Never mind, I see someone already has:

https://github.com/hydrogen-music/hydrogen/issues/1309

**Comment by cme** [comment]
So, it looks like the Windows / MXE cross-compile scripting doesn't have any way of configuring LADSPA support (all that it really needs is to acquire the header from somewhere). So the most recent release builds haven't included it.

Previous release builds for Windows were built in @mauser's own environment so there may be an extra step that I didn't know about.

The new Appveyor builds seem to be able to get LADSPA so there shouldn't be a problem going forward.

**Reply by dfkettle** [reply]
Thanks! When will that be available, approximately?

**Comment by dfkettle** [comment]
Is there an earlier version of Hydrogen for Windows available to download that does support LADSPA plug-ins?

**Reply by theGreatWhiteShark** [reply]
AFAIK we have means to generate a windows binary for an arbitrary commit and why not use this for testing. But for some reason the Windows build does seem to have a problem recently (without us touching it). So, this needs to be fixed beforehand. I'll have a look.

**Comment by lacostenycoder** [comment]
Hey I'm on Linux. I built/installed these https://github.com/swh/ladspa and now I can see them in the effects and I can add them to the effects rack. However, they seem to have absolutely NO effect at all on the sound.  What am I missing?

**Reply by cme** [reply]
Possssibly that, to make anything audible from the effects, you have to select which instruments send their output to the effect unit, by tweaking one of these 'FX send' knobs in the mixer:

![image](https://user-images.githubusercontent.com/768466/206758336-2dfe8a42-58cc-4959-bddb-423eab92636e.png)

?

You'll still hear the unfiltered instrument as well, but turning down the slider knob on that will let you hear only the filtered sound.

**Reply by lacostenycoder** [reply]
@cme Thanks!  I see now that to make it act like an "insert" I have to kill the channel fader and boost full wet signal on the FX!  Cheers! 🍻
